backend/controllers/posts.go

```diff
@@ -25,8 +25,8 @@ func (con *PostsController) GetAllPosts(c echo.Context) error {
 func (con *PostsController) GetPost(c echo.Context) error {
 	var post models.Post
 
-	if result := con.DB.Find(&post, c.Param("id")); result.Error != nil {
-		return c.JSON(http.StatusInternalServerError, "error")
+	if result := con.DB.Take(&post, c.Param("id")); result.Error != nil {
```

Comment:
ここ、とかでもいけるんだろうけど、c.Bind(&post)でエラーがほとんど無さそう(というかいつ出るかわからん...)なのと、どうせpathのid間違っていたら`Take`でエラーになるんで、ここはpath paramでデータ抽出する方がいいと思いましたまる。
```suggestion
	if err := c.Bind(&post); err != nil {
        	return c.JSON(http.StatusBadRequest, "bad request")
	}

	if result := con.DB.Take(&post); result.Error != nil {
		...
	}
```

Reply:
ただ、なんでTakeは
err := con.DB.Take(&post)じゃだめなんだ！？
と思ったけど、定義見てみたら、*DBを返すってなってた。
返り値が errorだったら `err := ***` でいけるって認識であってます？